components/did/Core/Src/DID.c

```diff
@@ -250,22 +249,32 @@ void update_DID_screen()
 				OutputString("---", STATE_DATA_XPOS, STATE_DATA_YPOS);
 			}
 
-			// Target cruise
-			if ( target_velocity != NULL && MCB_drive_state != NULL )
+			// Regen Enable
+			if ( MCB_regen_enabled != NULL )
 			{
-				if ( *MCB_drive_state == 0x02 )
-				{
-					UpdateScreenParameter(CRUISE_DATA_XPOS, CRUISE_DATA_YPOS, (uint32_t)(*target_velocity), 0, FALSE);
-				}
-				else
-				{
-					OutputString("OFF", CRUISE_DATA_XPOS, CRUISE_DATA_YPOS);
+				if (*MCB_regen_enabled == 0x01) {
+					OutputString("   ", REGEN_DATA_XPOS, REGEN_DATA_YPOS);
+					OutputString("ON", REGEN_DATA_XPOS, REGEN_DATA_YPOS);
+				} else {
+					OutputString("   ", REGEN_DATA_XPOS, REGEN_DATA_YPOS);
+					OutputString("OFF", REGEN_DATA_XPOS, REGEN_DATA_YPOS);
 				}
 			}
 			else
 			{
-				OutputString("---", CRUISE_DATA_XPOS, CRUISE_DATA_YPOS);
+				OutputString("---", REGEN_DATA_XPOS, REGEN_DATA_YPOS);
```

Comment:
Consider adding a define for "---" incase we change what to display in the future

Reply:
Sure